### ISSUE TITLE
希望persist支持传入数组 (Allow different storages to be used in a single store)

### ISSUE THREAD

**Issue by zhuzhengshou** [prazdevs/pinia-plugin-persistedstate#19]
你好，希望作者考虑一下defineStore时，传入options中的presist属性，支持PersistedStateOptions[]数组传入。
因为有时希望一个store中的数据可以存储在不同的storage

**Comment by prazdevs** [comment]
Hi thanks for opening the issue. I didn't specify it in the guideline or such, but I don't speak any Chinese :/ Could you rephrase the whole thing in English so I can help you ? 

Thanks!

**Comment by zhuzhengshou** [comment]
sorry，It's my thoughtlessness

**Comment by zhuzhengshou** [comment]
sorry, It's my thoughtlessness!
well, I mean this tool is very good。so that I hope it can be better。
for example：
 I hope  when I use defineStore of pinia , based on your expanded parameter -- persist，Its value can be an array(persist?: boolean | PersistedStateOptions | PersistedStateOptions[])。
because, Sometimes，all data in one store, hope can be persisted in different storage(localStorage, sessionStorage)
```
export const userStore = defineStore('user', {
  state: () => {
    return {
      key1: '',
      key2: ''
    }
  },
  persist: [
    {
      key: 'piniaUserStoreInSession',
      storage: window.sessionStorage,
      paths: ['key1']
    },
    {
      key: 'piniaUserStoreInLocal',
      storage: window.localStorage,
      paths: ['key2']
    }
  ]
})
```
The above is through translation software, If there is any inaccuracy, I hope you will forgive me!

**Comment by prazdevs** [comment]
Ooooh I see it now, the example code helped a lot. The idea seems interesting. I'll consider it and try to implement this when i get some time. I'm just a bit concerned with performance but that would be documented.

(no worries as for English, it's more than i need to understand your problem!)

**Comment by prazdevs** [comment]
Hi, as I thought about implementing this, I came across an issue.
If a same paths is stored in different stores, how about rehydration? That would let the possibility to have 2 different sources of rehydration 🤔

**Comment by zhuzhengshou** [comment]
> Hi, as I thought about implementing this, I came across an issue. If a same paths is stored in different stores, how about rehydration? That would let the possibility to have 2 different sources of rehydration 🤔

It's nice to hear from you again.
can you list some scenes at work?
because, I think nobody use like this（a same paths is stored in different stores).
if It won't be used like that, maybe can throw a error to prevent the problem？

**Comment by N0tExisting** [comment]
@zhuzhengshou or @prazdevs could one of you please change the issue title to English?

**Comment by prazdevs** [comment]
Finally implemented and released it in V2 🎉 
Closing this !

**Comment by CodeListener** [comment]
@prazdevs  @zhuzhengshou  hi, I don't understand why we need to support arrays. Can you provide a relevant use case ?
